asyncpg/pool.py

```diff
@@ -566,7 +566,7 @@ async def fetchrow(self, query, *args, timeout=None):
         async with self.acquire() as con:
             return await con.fetchrow(query, *args, timeout=timeout)
 
-    def acquire(self, *, timeout=None):
+    def acquire(self, *, timeout=None) -> connection.Connection:
```

Comment:
This type hint only covers usual `await` case:

```python
con = await pool.acquire()
```

But won't work properly for context manager case:

```python
async with pool.acquire() as con:
    await con.execute(...)
```

The more correct return type could be something like `PoolAcquireContext` but we should check if `mypy` understands it first.

Reply:
#577 contains more precise types